sklearn/metrics/tests/test_pairwise.py

```diff
@@ -445,6 +446,22 @@ def test_rbf_kernel():
     assert_array_almost_equal(K.flat[::6], np.ones(5))
 
 
+def test_cosine_similarity_sparse_output():
+    # Test if cosine_similarity correctly produces sparse output.
+
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((5, 4))
+    Y = rng.random_sample((3, 4))
+    Xcsr = csr_matrix(X)
+    Ycsr = csr_matrix(Y)
+
+    K1 = cosine_similarity(Xcsr, Ycsr, dense_output=False)
+    assert_true(isinstance(K1, spmatrix))
```

Comment:
You can use `issparse`:
http://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.sparse.issparse.html